src/stores/query-bar-store.js

```diff
@@ -713,8 +732,8 @@ const configureStore = (options = {}) => {
     localAppRegistry.on('subtab-changed', () => {
       options.actions.refreshEditor();
     });
-    localAppRegistry.on('compass:schema:geo-query', (query) => {
-      store.setQuery({ filter: query }, true);
+    localAppRegistry.on('compass:schema:geo-query', (filter) => {
+      store.mergeQuery({ filter }, true);
```

Comment:
Probably it is also possible to "cherry pick" the geo query and explicitly set only the geo query related stuff in the filter? Just asking since `mergeQuery` seems to touch also other part of the filter, maybe being more strict would help to prevent having unwanted side effects in future.

Reply:
Yeah, we could do that, but on the other hand if we got more filter properties sent from the schema plugin, we probably shouldn’t ignore that, right? I feel like that would be more surprising here ;)